floor_generator/lib/misc/extension/dart_object_extension.dart

```diff
@@ -1,14 +1,26 @@
 import 'package:analyzer/dart/constant/value.dart';
 import 'package:analyzer/dart/element/type.dart';
+import 'package:floor_annotation/floor_annotation.dart';
+import 'package:meta/meta.dart';
 
 extension DartObjectExtension on DartObject {
-  String toEnumValueString() {
+  /// get the String representation of the enum value, or the result of
+  /// [orElse] if the enum was not valid. [orElse]
+  String toEnumValueString({@required String orElse()}) {
     final interfaceType = type as InterfaceType;
     final enumValue = interfaceType.element.fields
         .where((element) => element.isEnumConstant)
         .map((fieldElement) => fieldElement.name)
-        .singleWhere((valueName) => getField(valueName) != null);
+        .singleWhere((valueName) => getField(valueName) != null,
+            orElse: orElse);
 
     return '$interfaceType.$enumValue';
   }
+
+  ForeignKeyAction toForeignKeyAction({@required ForeignKeyAction orElse()}) {
+    final enumValueString = toEnumValueString(orElse: () => null);
```

Comment:
💡 As when the `orElse` value is used here, we could return early and just use the value produced from the `orElse` function parameter.

```suggestion
    final enumValueString = toEnumValueString(orElse: () => null);
    if (enumValueString == null) return orElse();
```

Reply:
I don't think `enumValueString` can ever be `null`, because `toEnumValueString` will always return `'$interfaceType.$enumValue'`, or in the orElse case: `'$interfaceType.null'`. So this short-circuit won't work.

Reply:
But I see that this `orElse` parameter in `toEnumValueString` is misleading. I should change that.